shoop/admin/modules/orders/static_src/create/reducers/lines.js

```diff
@@ -111,33 +124,44 @@ function setLineProperty(state, {payload}) {
                 updates.type = value;
                 break;
             case "quantity":
-                var quantity = Math.max(0, parseFloat(value));
+                const quantity = Math.max(0, ensureNumericValue(value, 1));
                 updates = getDiscountsAndTotal(quantity, line.baseUnitPrice, line.unitPrice);
                 updates.quantity = quantity;
                 break;
             case "unitPrice":
-                var unitPrice = parseFloat(value);
-                updates = getDiscountsAndTotal(line.quantity, line.baseUnitPrice, unitPrice);
-                updates.unitPrice = unitPrice;
+                updates = getDiscountsAndTotal(
+                    line.quantity,
+                    line.baseUnitPrice,
+                    ensureNumericValue(value, line.baseUnitPrice),
+                    true
+                );
                 break;
             case "discountPercent":
-                var discountPercent = Math.max(0, parseFloat(value));
+                var discountPercent = ensureNumericValue(value);
+                discountPercent = Math.max(0, discountPercent);
+                if (discountPercent > 100)
```

Comment:
Braces, please. Also, as you're using the idiom `Math.max` on the line above, just do

```javascript
const discountPercent = Math.min(100, Math.max(0, ensureNumericValue(discountPercent)));
```